pkg/serverless/appsec/httpsec/http_test.go

```diff
@@ -86,3 +106,19 @@ func getEventFromFile(filename string) []byte {
 	buf.WriteString("0")
 	return buf.Bytes()
 }
+
+func TestInvocationSubProcessorNilInterface(t *testing.T) {
+	lp := &invocationlifecycle.LifecycleProcessor{
+		DetectLambdaLibrary: func() bool { return true },
+		SubProcessor:        (*httpsec.InvocationSubProcessor)(nil),
+	}
+
+	assert.True(t, lp.SubProcessor != nil)
+
+	lp.OnInvokeStart(&invocationlifecycle.InvocationStartDetails{
+		InvokeEventRawPayload: []byte(
+			`{"requestcontext":{"stage":"purple"},"httpmethod":"purple","resource":"purple"}`),
+	})
+
+	lp.OnInvokeEnd(&invocationlifecycle.InvocationEndDetails{})
+}
```

Comment:
I'd suggest you move this test into the lifecycle processor test suite as this is testing its nil-check is working as expected.

Reply:
Good catch.